src/Fahrplanauskunft/Fahrplanauskunft/Funktionen/Logik.cs

```diff
@@ -80,5 +81,156 @@ internal static List<Umstiegspunkt> Liefere_Naechste_Umstiegspunkte_von_Halteste
 
             return umstiegspunkte;
         }
+
+        /// <summary>
+        /// Gibt die sortierte Liste von Haltestellen für eine Linie zurück. Sortiert wird sie von der Start- zur Ziel-Haltestelle
+        /// </summary>
+        /// <param name="linie">Die Linie, für die sortierung ausgeführt werden soll</param>
+        /// <param name="startHaltestelle">Die Start-Haltestelle, bei der die Sortierung beginnen soll</param>
+        /// <param name="zielHaltestelle">Die Ziel-Haltestelle, bei der die Sortierung enden soll</param>
+        /// <param name="haltenstellen">Das gesamte Haltestellennetz</param>
+        /// <param name="streckenabschnitte">Alle verfügbaren Streckenabschnitte</param>
+        /// <returns></returns>
+        internal static List<Haltestelle> Sortiere_Liste_von_Haltestellen_von_Start_nach_Ziel(Linie linie, Haltestelle startHaltestelle, Haltestelle zielHaltestelle, List<Haltestelle> haltenstellen, List<Streckenabschnitt> streckenabschnitte)
```

Comment:
# Too complex:
Die Funktion ist zu komplex.  
Hier sollte mit etwas Umgestalten die Komplexität herausgenommen werden.  
Potential liegt bestimmt in dem IF..ELSE Block  
```
Namespace: Fahrplanauskunft.Funktionen
Type: Logik
Member: Sortiere_Liste_von_Haltestellen_von_Start_nach_Ziel(Linie, Haltestelle, Haltestelle, List<Haltestelle>, List<Streckenabschnitt>) : List<Haltestelle>
Maintainability Index: 41
Cyclomatic Complexity: 15
Class Coupling: 10
Lines of Code: 41
```